addons/stock/models/stock_picking.py

```diff
@@ -566,6 +569,18 @@ def _compute_show_validate(self):
             else:
                 picking.show_validate = True
 
+    @api.depends('state', 'immediate_transfer', 'move_lines', 'move_lines.product_id.tracking', 'move_lines.quantity_done')
+    def _compute_show_set_all_done(self):
+        precision_digits = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+        for picking in self:
+            if (picking.state == 'draft' and picking.immediate_transfer) or picking.state in ('waiting', 'confirmed', 'assigned'):
```

Comment:
missing a not for the 1st part of condition no? I don't see why we would set all done quantities for a picking w/h initial demand.